### PR TITLE
feat: read/write works in distributed mode 🎉

### PR DESCRIPTION
**Pull request by waynexia**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

The last piece of new distributed mode

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/api/src/helper.rs [hunk 1]
```diff
@@ -837,10 +837,10 @@ pub fn value_to_grpc_value(value: Value) -> GrpcValue {
             Value::Date(v) => Some(ValueData::DateValue(v.val())),
             Value::DateTime(v) => Some(ValueData::DatetimeValue(v.val())),
             Value::Timestamp(v) => Some(match v.unit() {
-                TimeUnit::Second => ValueData::TimeSecondValue(v.value()),
-                TimeUnit::Millisecond => ValueData::TimeMillisecondValue(v.value()),
-                TimeUnit::Microsecond => ValueData::TimeMicrosecondValue(v.value()),
-                TimeUnit::Nanosecond => ValueData::TimeNanosecondValue(v.value()),
+                TimeUnit::Second => ValueData::TsSecondValue(v.value()),
```
Comment (v0y4g3r) [inline src/api/src/helper.rs:840]: This naming is annoying.

File: src/client/src/region.rs [hunk 2]
```diff
@@ -54,6 +63,76 @@ impl RegionRequester {
...
+                    tonic_code,
+                    addr: flight_client.addr().to_string(),
+                    source: BoxedError::new(ServerSnafu { code, msg }.build()),
+                };
+                error!(
+                    "Failed to do Flight get, addr: {}, code: {}, source: {}",
+                    flight_client.addr(),
+                    tonic_code,
+                    error
+                );
```
Comment (evenyag) [inline src/client/src/region.rs:87]:
```suggestion
                error!(
                    e; "Failed to do Flight get, addr: {}, code: {}",
                    flight_client.addr(),
                    tonic_code
                );
```

**Review by v0y4g3r — APPROVED** [review]
LGTM

File: src/client/src/region.rs [hunk 3]
```diff
@@ -54,6 +63,76 @@ impl RegionRequester {
...
+            .fail();
+        };
+        let FlightMessage::Schema(schema) = first_flight_message? else {
+            return IllegalFlightMessagesSnafu {
+                reason: "Expect schema to be the first flight message",
+            }
+            .fail();
+        };
+
+        let stream = Box::pin(stream!({
```
Comment (evenyag) [inline src/client/src/region.rs:113]: Could we use try_stream?

File: src/client/src/region.rs [hunk 4]
```diff
@@ -54,6 +63,76 @@ impl RegionRequester {
...
+            while let Some(flight_message) = flight_message_stream.next().await {
+                let flight_message = flight_message
+                    .map_err(BoxedError::new)
+                    .context(ExternalSnafu)?;
+                let FlightMessage::Recordbatch(record_batch) = flight_message else {
+                    yield IllegalFlightMessagesSnafu {reason: "A Schema message must be succeeded exclusively by a set of RecordBatch messages"}
+                                        .fail()
+                                        .map_err(BoxedError::new)
+                                        .context(ExternalSnafu);
```
Comment (evenyag) [inline src/client/src/region.rs:122]: Use context to create IllegalFlightMessages error.

Comment (waynexia) [reply src/client/src/region.rs:122]: ?

File: src/api/src/helper.rs (hunk 1 above)
Comment (waynexia) [reply src/api/src/helper.rs:840]: 🥹

File: src/client/src/region.rs (hunk 3 above)
Comment (waynexia) [reply src/client/src/region.rs:113]: they are identical

**Review by evenyag — APPROVED** [review]
kbd

**Pull request merged**